tooling/bundler/src/bundle.rs

```diff
@@ -43,25 +42,38 @@ pub fn bundle_project(settings: Settings) -> crate::Result<Vec<Bundle>> {
   let mut bundles = Vec::new();
   let package_types = settings.package_types()?;
 
+  let target_os = settings
+    .target()
+    .split('-')
+    .nth(2)
+    .unwrap_or(std::env::consts::OS)
+    .replace("darwin", "macos");
+
+  if target_os != std::env::consts::OS {
+    warn!("Cross-platform compilation is experimental and does not support all features. Please use a matching host system for full compatibility.");
```

Comment:
A link to documentation which describes what exactly is not supported would be great.

Reply:
Is it only `updates` and `signing`?

Reply:
> A link to documentation which describes what exactly is not supported would be great.

There are no docs yet, but i guess a link to https://tauri.app/v1/guides/building/cross-platform should be a fairly safe bet.
"What exactly is not supported" - That's actually a bit tricky to know upfront. Obvious candidates are code signing and the .msi bundle target.
The idea of that message is more like "This is experimental and you should prefer native hosts, we don't know what works yet either" etc. Also i'd like to have it give us a bit more room to add features that only works if you compile it on the native host without that being a huge surprise to users.